### PR TITLE
Added support for ceil mode in AvgPool

### PR DESCRIPTION
**Pull request by jbobba**

### CODE REVIEW

File: src/ngraph/op/avg_pool.cpp [hunk 1]
```diff
@@ -314,6 +329,11 @@ shared_ptr<Node> op::AvgPoolBackprop::copy_with_new_args(const NodeVector& new_a
 
 void op::AvgPool::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
 {
+    if (m_ceil_mode)
+    {
+        throw ngraph_error("Autodiff not supported on AvgPool with ceil_mode set");
```
Comment (aprocter) [inline src/ngraph/op/avg_pool.cpp]: Can we open a JIRA for this? Shouldn't be too hard to do make it work.

File: test/type_prop.cpp [hunk 2]
```diff
@@ -8647,6 +8647,27 @@ TEST(type_prop, avg_pool_3d_deduce_strided_padded_small)
     EXPECT_EQ(avg_pool->get_padding_above(), (Shape{6, 4, 5}));
 }
 
+TEST(type_prop, avg_pool_ceil_mode)
```
Comment (aprocter) [inline test/type_prop.cpp]: We'll want some execution tests ASAP as well. Currently BEs will be silently ignoring this flag.

**Comment by ilyachur** [comment]
Could you add support for ceil mode in MaxPool too?

File: test/type_prop.cpp (hunk 2 above)
Comment (jbobba) [reply test/type_prop.cpp]: i was thinking of modifying the interpreter kernel but wasn't sure how to handle the extra rows/columns (ignore/include) Will need to look at the frameworks.

**Pull request merged**